### PR TITLE
Refactor chain service with block service

### PR DESCRIPTION
**Pull request by MichaelKostroma**

### CODE REVIEW

File: background/services/block/index.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        this.pollBlockPricesForNetwork(network.chainID)
+      )
+    )
+  }
+
+  async getBlockPrices(
+    network: NetworkInterface,
+    provider: JsonRpcProvider,
+    shard: Shard,
+    zone: Zone
```
Comment (rileystephens28) [inline background/services/block/index.ts]: A better solution here would be to only accept `shard` as an input and cast it to `zone` using the `toZone` method.

File: background/services/block/db.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import Dexie, { DexieOptions } from "dexie"
+
+import { AnyEVMBlock } from "../../networks"
+import { NetworkInterface } from "../../constants/networks/networkTypes"
+import { ChainDatabase } from "../chain/db"
```
Comment (rileystephens28) [inline background/services/block/db.ts]: Remove unused import

File: background/services/chain/index.ts [hunk 3]
```diff
@@ -906,7 +803,8 @@ export default class ChainService extends BaseService<Events> {
     const networkWasInactive = this.networkIsInactive(chainID)
     this.lastUserActivityOnNetwork[chainID] = Date.now()
     if (networkWasInactive) {
-      this.pollBlockPricesForNetwork(chainID)
+      // TODO
+      // await this.blockService.pollBlockPricesForNetwork(chainID)
     }
```
Comment (rileystephens28) [inline background/services/chain/index.ts]: Do we plan on doing anything here in the future?

File: background/services/block/db.ts (hunk 2 above)
Comment (MichaelKostroma) [reply background/services/block/db.ts]: Done

File: background/services/block/index.ts (hunk 1 above)
Comment (MichaelKostroma) [reply background/services/block/index.ts]: Thank you for the suggestion! I’ve implemented `toZone` method. This approach indeed simplifies the logic and improves the solution.

Feel free to review the updated code and let me know if you have any further feedback.

File: background/services/chain/index.ts (hunk 3 above)
Comment (MichaelKostroma) [reply background/services/chain/index.ts]: Thank you for asking! I’ve removed this function and left only the necessary functionality as it’s not currently needed.

**Review by rileystephens28 — APPROVED** [review]

**Pull request merged**